doc/source/reshaping.rst

```diff
@@ -228,6 +228,24 @@ which level in the columns to stack:
    df2.stack('exp')
    df2.stack('animal')
 
+.. versionadded: 0.18.0
+
+Unstacking can result in missing values if subgroups do not have the same
+set of labels.  By default, missing values will be replaced with NaN.
```

Comment:
double back-ticks around ``NaN`` (mention that for datetimelike these will be filled with ``NaT``), so maybe say the fill value for that dtype